src/test_typing_extensions.py

```diff
@@ -5779,6 +5779,33 @@ class D(Protocol[T1, T2, Unpack[Ts]]): pass
                     with self.assertRaises(TypeError):
                         klass[int]
 
+    def test_with_non_typevartuple(self):
+        # The two functions below are used by typing._GenericAlias and typing.Generic via monkey patching.
```

Comment:
Could you add a test that exercises public functionality? `_collect_type_vars` is a private function and it doesn't matter what it does; we're interested in something like `MyTypedDictT[int]` misbehaving.

It's fine to keep the tests you have here as more focused regression tests.